engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheHashingJobActorSpec.scala

```diff
@@ -0,0 +1,292 @@
+    // will stay Some(...)
+    val cchja = makeCCHJA(Option(TestProbe().ref), fileHashingActor.ref, TestProbe().ref, writeToCache = true, Option(newData -> Option(result)))
+    watch(cchja)
+    cchja.setState(HashingFiles)
+
+    cchja ! FileHashResponse(mock[HashResult])
+
+    // This proves that the ccjha keeps hashing files even though there is no ccreader requesting more hashes
+    fileHashingActor.expectMsg(fileHashRequest)
+    cchja.stateName shouldBe HashingFiles
+  }
+
+  it should "send FinalFileHashingResult to parent and CCReader and die" in {
+    val parent = TestProbe()
+    val callCacheReadProbe = TestProbe()
+    List(CompleteFileHashingResult(Set(mock[HashResult]), "AggregatedFileHash"), NoFileHashesResult) foreach { result =>
+      val newData = CallCacheHashingJobActorData(List.empty, List.empty, Option(callCacheReadProbe.ref))
+      val cchja = makeCCHJA(Option(callCacheReadProbe.ref), TestProbe().ref, parent.ref, writeToCache = true, Option(newData -> Option(result)))
+
+      parent.expectMsgClass(classOf[InitialHashingResult])
+      callCacheReadProbe.expectMsgClass(classOf[InitialHashingResult])
+
+      parent.watch(cchja)
+      cchja.setState(HashingFiles)
+
+      cchja ! FileHashResponse(mock[HashResult])
+
+      callCacheReadProbe.expectMsg(result)
+      parent.expectMsg(result)
+      parent.expectTerminated(cchja)
+    }
+  }
+
+  it should "wait for next file hash if the batch is not complete yet" in {
+    val callCacheReadProbe = TestProbe()
+    val parent = TestProbe()
+    val newData: CallCacheHashingJobActorData = CallCacheHashingJobActorData(List.empty, List.empty, Option(callCacheReadProbe.ref))
+    val cchja = makeCCHJA(Option(callCacheReadProbe.ref), TestProbe().ref, parent.ref, writeToCache = true, Option(newData -> None))
+
+    parent.expectMsgClass(classOf[InitialHashingResult])
+    callCacheReadProbe.expectMsgClass(classOf[InitialHashingResult])
+
+    cchja.setState(HashingFiles)
+
+    cchja ! FileHashResponse(mock[HashResult])
+
+    callCacheReadProbe.expectNoMsg()
+    parent.expectNoMsg()
+    cchja.stateName shouldBe HashingFiles
+  }
+
+  it should "stop if the read actor dies and writeToCache is off" in {
+    val callCacheReadProbe = TestProbe()
+    val parent = TestProbe()
+    val cchja = makeCCHJA(Option(callCacheReadProbe.ref), TestProbe().ref, parent.ref, writeToCache = false)
+    parent.watch(cchja)
+    system stop callCacheReadProbe.ref
+    parent.expectMsgClass(classOf[InitialHashingResult])
+    parent.expectTerminated(cchja)
+  }
+
+  it should "keep going if the read actor dies and writeToCache is on" in {
+    val callCacheReadProbe = TestProbe()
+    val parent = TestProbe()
+    val cchja = makeCCHJA(Option(callCacheReadProbe.ref), TestProbe().ref, parent.ref)
+    parent.expectMsgClass(classOf[InitialHashingResult])
+    
+    cchja.setState(WaitingForHashFileRequest)
+    system stop callCacheReadProbe.ref
+
+    parent.expectMsg(NoFileHashesResult)
```

Comment:
It'd be nice if we could have some file hashes here (to make sure it did indeed keep going)